wings/tracker/bamboo_relay_order_book_tracker.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python
+
+import asyncio
+import bisect
+from collections import deque, defaultdict
+import logging
+import time
+from typing import (
+    Deque,
+    Dict,
+    List,
+    Optional,
+    Set
+)
+
+from wings.data_source.bamboo_relay_local_cluster_order_book_data_source import BambooRelayLocalClusterOrderBookDataSource
+from wings.order_book_tracker import OrderBookTracker, OrderBookTrackerDataSourceType
+from wings.data_source.order_book_tracker_data_source import OrderBookTrackerDataSource
+from wings.data_source.bamboo_relay_api_order_book_data_source import BambooRelayAPIOrderBookDataSource
+from wings.order_book_message import OrderBookMessageType, BambooRelayOrderBookMessage
+from wings.order_book_tracker_entry import BambooRelayOrderBookTrackerEntry
+from wings.orderbook.bamboo_relay_order_book import BambooRelayOrderBook
+from wings.tracker.bamboo_relay_active_order_tracker import BambooRelayActiveOrderTracker
+
+
+class BambooRelayOrderBookTracker(OrderBookTracker):
+    _rrobt_logger: Optional[logging.Logger] = None
+
+    @classmethod
+    def logger(cls) -> logging.Logger:
+        if cls._rrobt_logger is None:
+            cls._rrobt_logger = logging.getLogger(__name__)
+        return cls._rrobt_logger
+
+    def __init__(self,
+                 data_source_type: OrderBookTrackerDataSourceType = OrderBookTrackerDataSourceType.EXCHANGE_API,
+                 symbols: Optional[List[str]] = None):
+        super().__init__(data_source_type=data_source_type)
+
+        self._ev_loop: asyncio.BaseEventLoop = asyncio.get_event_loop()
+        self._data_source: Optional[OrderBookTrackerDataSource] = None
+        self._order_book_snapshot_stream: asyncio.Queue = asyncio.Queue()
+        self._order_book_diff_stream: asyncio.Queue = asyncio.Queue()
+        self._process_msg_deque_task: Optional[asyncio.Task] = None
+        self._past_diffs_windows: Dict[str, Deque] = {}
+        self._order_books: Dict[str, BambooRelayOrderBook] = {}
+        self._saved_message_queues: Dict[str, Deque[BambooRelayOrderBookMessage]] = defaultdict(lambda: deque(maxlen=1000))
+        self._active_order_trackers: Dict[str, BambooRelayActiveOrderTracker] = defaultdict(BambooRelayActiveOrderTracker)
+        self._symbols: Optional[List[str]] = symbols
+
+    @property
+    def data_source(self) -> OrderBookTrackerDataSource:
+        if not self._data_source:
+            if self._data_source_type is OrderBookTrackerDataSourceType.EXCHANGE_API:
+                self._data_source = BambooRelayAPIOrderBookDataSource(symbols=self._symbols)
+            elif self._data_source_type is OrderBookTrackerDataSourceType.LOCAL_CLUSTER:
```

Comment:
just raise an error if local cluster type is selected.

Reply:
You can reference Coinbase Pro's order book tracker for how it's done.

Reply:
Sure, just removed the local cluster option.